components/style/viewport.rs

```diff
@@ -220,6 +325,149 @@ impl ViewportRule {
 
         Ok(ViewportRule { declarations: valid_declarations.iter().cascade() })
     }
+
+    pub fn from_meta<'a>(content: &'a str) -> Option<ViewportRule> {
+        let mut iter = content.chars().enumerate();
+
+        macro_rules! start_of_name {
+            () => {
```

Comment:
I think I'd prefer passing in `iter` as an argument to the macro to make it clear what variable it's using.